scopes/preview/preview/pre-bundle-preview.task.ts

```diff
@@ -0,0 +1,108 @@
+import { join } from 'path';
+import { existsSync, mkdirSync, writeFileSync } from 'fs';
+import { BuildContext, BuildTask, BuiltTaskResult, TaskLocation } from '@teambit/builder';
+import { Capsule } from '@teambit/isolator';
+import { Logger } from '@teambit/logger';
+import { UIRoot, UiMain } from '@teambit/ui';
+import { PreviewAspect } from '@teambit/preview';
+import { promisify } from 'util';
+import webpack from 'webpack';
+import { sha1 } from '@teambit/legacy/dist/utils';
+import createPreBundleConfig from './webpack/webpack.prebundle.config';
+import { getEntryForPreBundlePreview } from './pre-bundle-preview';
+
+export const PRE_BUNDLE_PREVIEW_TASK_NAME = 'PreBundlePreview';
+export const PRE_BUNDLE_PREVIEW_DIR = 'preview-pre-bundle';
+export const PRE_BUNDLE_PREVIEW_HASH_FILENAME = '.hash';
+
+async function build(uiMain: UiMain, logger: Logger, outputPath: string): Promise<webpack.Stats | undefined> {
+  logger.debug(`pre-bundle for preview: start`);
+  const ui = uiMain.getUi();
+  if (!ui) throw new Error('ui not found');
+  const [rootExtensionName, uiRoot] = ui;
+  const resolvedAspects = await uiRoot.resolveAspects('preview');
+  const mainEntry = getEntryForPreBundlePreview(resolvedAspects, rootExtensionName, 'preview', PreviewAspect.id);
```

Comment:
I wonder about the fact that we pass the ui aspect here in the second arg. 
Maybe we should change it. but I need to better understand it.

Reply:
It's a replacement of this line:
https://github.com/teambit/bit/blob/9cd1104a42f0dcd744d454e56d4908642ad4c6c0/scopes/preview/preview/preview.main.runtime.tsx#L662

and the `name` is from:
https://github.com/teambit/bit/blob/9cd1104a42f0dcd744d454e56d4908642ad4c6c0/scopes/preview/preview/preview.main.runtime.tsx#L659